medallion/test/test_memory_backend.py

```diff
@@ -372,3 +376,190 @@ def test_saving_data_file(self):  # just for the memory backend
+        )
+        self.assertEqual(r.status_code, 401)
+
+    def test_get_object_403(self):
+        """note that the 403 code is still being generated at the Collection resource level
+
+           (i.e. we dont have access rights to the collection specified, not just the object)
+        """
+        r = self.client.get(
+            "/trustgroup1/collections/64993447-4d7e-4f70-b94d-d7f33742ee63/objects/indicator--b81f86b9-975b-bb0b-775e-810c5bd45b4f/",
+            headers=self.auth
+        )
+        self.assertEqual(r.status_code, 403)
+
+    def test_get_object_404(self):
+        # TAXII spec allows for a 404 or empty bundle if object is not found
+        r = self.client.get(
+            "/trustgroup1/collections/91a7b528-80eb-42ed-a74d-c6fbd5a26116/objects/malware--cee60c30-a68c-11e3-b0c1-a01aac20d000/",
+            headers=self.auth
+        )
+        objs = self.load_json_response(r.data)
+
+        if r.status_code == 200:
+            assert len(objs["objects"]) == 0
+        else:
+            self.assertEqual(r.status_code, 404)
+
+    def test_get_or_add_objects_401(self):
+        # note that no credentials are supplied with requests
+
+        # get_objects()
+        r = self.client.get("/trustgroup1/collections/64993447-4d7e-4f70-b94d-d7f33742ee63/objects/")
+        self.assertEqual(r.status_code, 401)
+
+        # add_objects()
+        new_id = "indicator--%s" % uuid.uuid4()
+        new_bundle = copy.deepcopy(API_OBJECTS_2)
+        new_bundle["objects"][0]["id"] = new_id
+
+        post_header = {}
+        post_header["Content-Type"] = MEDIA_TYPE_STIX_V20
+        post_header["Accept"] = MEDIA_TYPE_TAXII_V20
+
+        r_post = self.client.post(
+            "/trustgroup1/collections/91a7b528-80eb-42ed-a74d-c6fbd5a26116/objects/",
+            data=json.dumps(new_bundle),
+            headers=post_header
+        )
+        self.assertEqual(r_post.status_code, 401)
+
+    def test_get_or_add_objects_403(self):
+        """note that the 403 code is still being generated at the Collection resource level
+
+           (i.e. we dont have access rights to collection specified here, not just the objects)
+        """
+        # get_objects()
+        r = self.client.get(
+            "/trustgroup1/collections/64993447-4d7e-4f70-b94d-d7f33742ee63/objects/",
+            headers=self.auth
+        )
+        self.assertEqual(r.status_code, 403)
+
+        # add_objects
+        new_id = "indicator--%s" % uuid.uuid4()
+        new_bundle = copy.deepcopy(API_OBJECTS_2)
+        new_bundle["objects"][0]["id"] = new_id
+
+        post_header = copy.deepcopy(self.auth)
+        post_header["Content-Type"] = MEDIA_TYPE_STIX_V20
+        post_header["Accept"] = MEDIA_TYPE_TAXII_V20
+
+        r_post = self.client.post(
+            "/trustgroup1/collections/64993447-4d7e-4f70-b94d-d7f33742ee63/objects/",
```

Comment:
It might be good to define some of these URLs as module-level variables so it's more clear which tests are related and how. Maybe something like [this](https://github.com/oasis-open/cti-taxii-client/blob/master/taxii2client/test/test_client.py#L15-L28)?